lib/sqlalchemy/orm/context.py

```diff
@@ -1292,7 +1292,15 @@ def _legacy_join(self, args):
                     if of_type:
                         right = of_type
                     else:
-                        right = onclause.property.entity
+                        right = onclause.property
+
+                        try:
+                            right = right.entity
+                        except AttributeError as err:
+                            raise sa_exc.ArgumentError(
```

Comment:
**mike bayer** (zzzeek) wrote:

use:

    util.raise_(sa_exc.ArgumentError(...), replace_context=err)

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2099

Reply:
I have made the changes now, thanks for the review. I do have a quick question in Jenkins
[Build 5278] (https://jenkins.sqlalchemy.org/job/sqlalchemy_gerrit/5278/) is my understanding correct that additional tests are ran to see if the code change works on all the supported database systems as well as on both python 2 and 3?